### PR TITLE
1.13 compatibility

### PR DESCRIPTION
**Pull request by Deudly**
- [x] Make diamond-square capable of generating 1.13 maps
- [x] Backwards compatibility

### CODE REVIEW

File: diamond_square.js [hunk 1]
```diff
@@ -3,6 +3,8 @@
 const Vec3 = require('vec3').Vec3
 const rand = require('random-seed')
 
+var theFlattering = [ "1.13", "1.14", "1.15", "1.16"]
```
Comment (rom1504) [inline diamond_square.js]: Flatening

**Review by rom1504 — CHANGES_REQUESTED** [review]

File: diamond_square.js [hunk 2]
```diff
@@ -101,31 +104,47 @@ function generation ({ version = '1.8', seed, worldHeight = 80, waterline = 20 }
...
-          else if (y < level && y >= dirtheight) block = belowblock // Dirt/sand below surface
-          else if (y < level) block = 1 // Set stone inbetween
-          else if (y === level) block = surfaceblock // Set surface sand/grass
-          else if (y <= waterline) block = 9 // Set the water
+          const surfaceblock = level < waterline ? mcData.blocksByName.sand.id : isFlatteningVersion(version) ?  mcData.blocksByName.grass_block.id :  mcData.blocksByName.grass.id // Sand below water, grass
+          const belowblock = level < waterline ? mcData.blocksByName.sand.id : mcData.blocksByName.dirt.id // 3-5 blocks below surface
+          if (y < bedrockheight) block = mcData.blocksByName.bedrock.id // Solid bedrock at bottom
+          else if (y < level && y >= dirtheight) {
+            block = belowblock // Dirt/sand below surface
+            if (isFlatteningVersion()) {
```
Comment (rom1504) [inline diamond_square.js]: you don't pass the version => this cannot work

**Review by rom1504 — CHANGES_REQUESTED** [review]
there's a small error to fix
+ could you test it with another version to check it still works there ?

**Comment by Deudly** [comment]
I've now tested it in 1.12.2 and it's working

**Comment by Deudly** [comment]
Okay, everything is tested now. Sorry for the commits spam. You should probably squash and merge

File: diamond_square.js [hunk 3]
```diff
@@ -3,6 +3,8 @@
 const Vec3 = require('vec3').Vec3
 const rand = require('random-seed')
 
+const theFlattening = [ "1.13.2", "1.14", "1.15", "1.16"]
```
Comment (rom1504) [inline diamond_square.js]: Use major versions

File: diamond_square.js [hunk 4]
```diff
@@ -101,31 +104,54 @@ function generation ({ version = '1.8', seed, worldHeight = 80, waterline = 20 }
...
+            if (isFlatteningVersion(version)) {
+              if (level < waterline) data = 0 // Default sand data is 0
+              else data = 1 // Default dirt data is 1, 0 is snowy
+            }
+          }
+          else if (y <= waterline) block = mcData.blocksByName.water.id // Set the water
           else if (y === level + 1 && level >= waterline && seedRand(10) === 0) { // 1/10 chance of grass
-            block = 31
-            data = 1
+            if (isFlatteningVersion(version)) {
```
Comment (rom1504) [inline diamond_square.js]: Better retrieve the major version from minecraft data (.version.majorVersion)

**Review by rom1504 — CHANGES_REQUESTED** [review]

**Comment by Deudly** [comment]
Everything should be okay now

**Review by rom1504 — APPROVED** [review]

**Pull request merged**